src/main/java/apoc/coll/Coll.java

```diff
@@ -127,6 +127,53 @@ public long indexOf(@Name("coll") List<Object> coll, @Name("value") Object value
     }
 
     @UserFunction
+    @Description("apoc.coll.set(coll, index, value) | set index to value")
+    public List<Object> set(@Name("coll") List<Object> coll, @Name("index") long index, @Name("value") Object value) {
+        if (coll == null) return null;
+        if (index < 0 || value == null) return coll;
+
+        List<Object> list = new ArrayList<>(coll);
+        list.set( (int) index, value );
+        return list;
+    }
+
+    @UserFunction
+    @Description("apoc.coll.insert(coll, index, value) | insert value at index")
+    public List<Object> insert(@Name("coll") List<Object> coll, @Name("index") long index, @Name("value") Object value) {
+        if (coll == null) return null;
+        if (index < 0 || value == null) return coll;
+
+        List<Object> list = new ArrayList<>(coll);
+        list.add( (int) index, value );
+        return list;
+    }
+
+    @UserFunction
+    @Description("apoc.coll.insertList(coll, index, values) | insert values at index")
+    public List<Object> insertList(@Name("coll") List<Object> coll, @Name("index") long index, @Name("values") List<Object> values) {
+        if (coll == null) return null;
+        if (index < 0 || values == null) return coll;
+
+        List<Object> list = new ArrayList<>(coll);
+        list.addAll( (int) index, values );
+        return list;
+    }
+
+    @UserFunction
+    @Description("apoc.coll.remove(coll, index, [length=1]) | remove range of values from index to length")
+    public List<Object> remove(@Name("coll") List<Object> coll, @Name("index") long index, @Name(value = "length") long length) {
+        if (coll == null) return null;
+        if (index < 0 || length < 0) return coll;
+
+        List<Object> list = new ArrayList<>(coll);
+        for (int i = 0; i < length; i++)
+        {
+            list.remove( (int) index );
```

Comment:
we should control that we don't exceed the length of the list? Or is that silently ignored?

Reply:
@jexp Don't you want users to get back raw error? Right now index out of bounds would just surface to the caller. Should we really silence that?

Reply:
Several Cypher operations on lists which exceed the index bounds are allowed, and don't throw exceptions. Seems like we should probably avoid this if possible.

Can we do a `length = Math.min(length, coll.size() - 1)` to prevent the exception?

Reply:
@InverseFalcon yeah i think that's reasonable if that's a standard way of handling out of bounds by user. most these decisions come down to uniform design being the best way. if cypher has a standard already I think we just follow that here